File: spring-webflux/src/main/java/victor/training/reactive/usecase/grouping/MessageType.java
package victor.training.reactive.usecase.grouping;

enum MessageType {
   TYPE1_NEGATIVE,
   TYPE2_ODD,
   TYPE3_EVEN;

   public static MessageType forMessage(Integer message) {
      if (message < 0) return TYPE1_NEGATIVE;
      if (message % 2 == 1) return TYPE2_ODD;
      else return TYPE3_EVEN;
   }
}

File: spring-webflux/src/main/java/victor/training/reactive/usecase/grouping/MessageType.kt
package victor.training.reactive.usecase.grouping

enum class MessageType {
    TYPE1_NEGATIVE, TYPE2_ODD, TYPE3_EVEN;

    companion object {
        fun forMessage(message: Int): MessageType {
            if (message < 0) return TYPE1_NEGATIVE
            return if (message % 2 == 1) TYPE2_ODD else TYPE3_EVEN
        }
    }
}